### PR TITLE
Expose current day as a parameter to showDatePicker.

### LINKED ISSUE #54526: showDatePicker should take the "current date" as an argument
When showing a DatePicker via the [showDatePicker function](https://api.flutter.dev/flutter/material/showDatePicker.html) it doesn't seem possible to influence the _current day_ (give it a different time zone for instance).

I'm in need of a DatePickerDialog that's completely handled in UTC (for aviation), ignoring the phone's current time zone.

Example:

My phone is set to CET and it's 07-APR-2020 00:30 CET, which without DST is 06-APR-2020 23:30 UTC. I've managed to show the currently selected date on the 6th by using the showDatePicker() function like this:

`showDatePicker(context: context, initialDate: DateTime.now().toUtc(), firstDate: DateTime.now().toUtc().subtract(Duration(days: 365)), lastDate: DateTime.now().toUtc().add(Duration(days: 90)));`

but the "current date" in the DayPicker is still in CET, demonstrated in this picture:

SDP

flutter doctor:
[✓] Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.15.4 19E287, locale en-AT)
    • Flutter version 1.12.13+hotfix.9 at /Users/egon/Development/flutter
    • Framework revision f139b11009 (11 days ago), 2020-03-30 13:57:30 -0700
    • Engine revision af51afceb8
    • Dart version 2.7.2

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/egon/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] Connected device (3 available)
    • Android SDK built for x86 • emulator-5554                        • android-x86 • Android 10 (API 29) (emulator)
    • iPhone 8 Plus             • CDFA08C8-A665-4EC3-9D19-20AED09E74F1 • ios         • com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)
    • iPhone 11 Pro Max         • 0E1F51E2-A165-4F9D-8B11-99644661E773 • ios         • com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)

• No issues found!

### PR DESCRIPTION
**Pull request by darrenaustin**
## Description

The current implementation of the Date Picker always displays the current day (denoted by a circular outline around the day) based on the device's current date (`DateTime.now()`). However, for apps that need to configure the current date, this PR adds a new `currentDate` parameter to the `showDatePicker` as well as the `CalendarDatePicker` component.

## Related Issues

Fixes: #54526

## Tests

I added a new test that checks to see if the provided current date is displayed highlighted.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them?

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/pickers/date_picker_dialog.dart [hunk 1]
```diff
@@ -93,6 +93,7 @@ Future<DateTime> showDatePicker({
   @required DateTime initialDate,
   @required DateTime firstDate,
   @required DateTime lastDate,
+  DateTime currentDate,
```
Comment (rami-a) [inline packages/flutter/lib/src/material/pickers/date_picker_dialog.dart:96]: Add some documentation above regarding this param.

**Review by rami-a — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/pickers/date_picker_dialog.dart (hunk 1 above)
Comment (darrenaustin) [reply packages/flutter/lib/src/material/pickers/date_picker_dialog.dart:100]: D'oh. I added it to `CalendarDatePicker` but not here. Fixed. Thx.

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**